tmexp/io_constants.py

```diff
@@ -1,5 +1,62 @@
+from typing import Counter as CounterType, Dict, List, NamedTuple, Optional, Set
+
 DATASET_DIR = "/data/datasets"
 
+
+class FileInfo(NamedTuple):
+    blob_hash: str
+    language: str
+
+
+class FilesInfo(dict):
+    def __init__(self, refs: List[str]):
+        super().__init__()
+        for ref in refs:
+            self[ref] = {}
+
+    def file_info(
+        self, ref: str, file_path: Optional[str] = None
+    ) -> Optional[FileInfo]:
+        return self[ref].get(file_path, None)
+
+    def add(self, ref: str, file_path: str, blob_hash: str, language: str) -> None:
+        self[ref][file_path] = FileInfo(blob_hash=blob_hash, language=language)
+
+    def remove(self, file_path: str, blob_hash: str) -> None:
+        for ref_dict in self.__dict__.values():
+            if file_path in ref_dict and blob_hash == ref_dict[file_path].blob_hash:
+                ref_dict.pop(file_path)
+
+
+WordCount = CounterType[str]
+FeatureContent = Dict[str, WordCount]
+BlobContent = Dict[str, FeatureContent]
+
+
+class FilesContent(dict):
```

Comment:
Should also subclass `Dict[str, FileContent]`

Reply:
*`Dict[str, BlobContent]` but yes totally, I'd forgotten here too